### PR TITLE
Dependent review component

### PR DESCRIPTION
**Pull request by amponce**
### Summary

This PR adds the ability for users to view and edit dependent ages on the review and submit page. If a user adds or removes a dependent, the corresponding dependent ages will become editable, allowing the user to adjust the age or add additional dependents.

Note: Once the feature is live, we will turn off the editing of dependent ages and display a read-only mode.

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#54028

## Testing done

- Manual testing
- Navigate to `/manage-va-debt/request-debt-help-form-5655/review-and-submit` after filling out the employment form to test adding and removing dependents.

### Desktop - Add/Edit Dependents (Veteran)

<table>
  <tr>
    <th>Dependent ages readOnly</th>
    <th>Dependents ages with edit</th>
    <th>Dependents ages editing</th>
  </tr>
  <tr>
 <td>
      Screenshot 1
    </td>
    <td>
      Screenshot 2
    </td>
    <td>
      Screenshot 3
    </td>
  </tr>
</table>

## What areas of the site does it impact?
Financial status report employment history form

## Acceptance criteria
- [x]  Dependents ages added to review and submit form.
- [x]  Unit and e2e test are updated and passing

### CODE REVIEW

File: src/applications/financial-status-report/sass/financial-status-report.scss [hunk 1]
```diff
@@ -3,7 +3,6 @@
 @import "~@department-of-veterans-affairs/formation/sass/modules/m-form-process";
 @import "~@department-of-veterans-affairs/formation/sass/modules/m-modal";
 @import "~@department-of-veterans-affairs/formation/sass/modules/m-omb-info";
-@import "~@department-of-veterans-affairs/formation/sass/shared-variables";
```
Comment (Dr-Pongo) [inline src/applications/financial-status-report/sass/financial-status-report.scss:6]: Why did we remove shared-variables here?

File: src/applications/financial-status-report/components/DependentAgesReview.jsx
Comment (Dr-Pongo) [inline src/applications/financial-status-report/components/DependentAgesReview.jsx:1]: There's a small part of me that would want to just include this in the same file as `DependentAges` bc it's kind of just a wrapper, but no need to change!

**Review by Dr-Pongo — APPROVED** [review]
LGTM! 🚀 
Nice work! I'd just change that `setData` import!

File: src/applications/financial-status-report/sass/financial-status-report.scss (hunk 1 above)
Comment (amponce) [reply src/applications/financial-status-report/sass/financial-status-report.scss:6]: It's a duplicate reference, that one is already defined on line 1
Screenshot 2023-05-05 at 8 40 36 AM

File: src/applications/financial-status-report/components/DependentAges.jsx [hunk 2]
```diff
@@ -1,11 +1,14 @@
-import React, { useState, useEffect, useMemo, useCallback } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
 import { useSelector, useDispatch } from 'react-redux';
 import { VaNumberInput } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
-import { setData } from 'platform/forms-system/src/js/actions';
+import { setData } from '@department-of-veterans-affairs/platform-forms-system/actions';
```
[the remark this answers is not in the document]
Comment (amponce) [reply src/applications/financial-status-report/components/DependentAges.jsx]: Fixed

**Pull request merged**